src/components/mainPage.js

```diff
@@ -1,59 +1,126 @@
-import React, { useState } from "react";
-import { makeStyles } from "@material-ui/core/styles";
-import AppBar from "@material-ui/core/AppBar";
-import Toolbar from "@material-ui/core/Toolbar";
-import Typography from "@material-ui/core/Typography";
-import IconButton from "@material-ui/core/IconButton";
-import Button from "@material-ui/core/Button";
-import MenuIcon from "@material-ui/icons/Menu";
+import { useState } from "react";
+import AppBar from "@mui/material/AppBar";
+import Toolbar from "@mui/material/Toolbar";
+import Typography from "@mui/material/Typography";
+import IconButton from "@mui/material/IconButton";
+import Button from "@mui/material/Button";
+import MenuIcon from "@mui/icons-material/Menu";
+import Menu from "@mui/material/Menu";
+import MenuItem from "@mui/material/MenuItem";
+import { useTheme } from "@mui/material/styles";
 import PotentialTable from "./potentialTable";
 import PercentageGraphs from "./percentageGraphs";
-
-import { useHistory } from "react-router-dom";
+import { Link, useNavigate } from "react-router-dom";
 import ShouldITodd from "./shouldITodd";
 
-const useStyles = makeStyles((theme) => ({
-  menuButton: {
-    marginRight: theme.spacing(2),
-  },
-  title: {
-    flexGrow: 1,
-  },
-  button: {
-    width: "120px",
-  },
-}));
+// const useStyles = makeStyles((theme) => ({
+//   menuButton: {
+//     marginRight: theme.spacing(2),
+//   },
+//   title: {
+//     flexGrow: 1,
+//   },
+//   button: {
+//     width: "120px",
+//   },
+// }));
 
 const MainPage = () => {
-  const classes = useStyles();
-  const history = useHistory();
+  // const classes = useStyles();
+  const navigate = useNavigate();
   const [open, setOpen] = useState(true);
+  // const [isMenuOpen, setMenuOpen] = useState(false);
+  const [anchorEl, setAnchorEl] = useState(null);
+  const theme = useTheme();
+  const styles = {
+    menuButton: {
+      marginRight: theme.spacing(2),
+    },
+    title: {
+      flexGrow: 1,
+    },
+    button: {
+      width: "120px",
+    },
+  };
 
   const handleOpen = () => {
     setOpen(!open);
   };
 
+  // const handleClose = () => {
+  //   setOpen(false);
+  // };
+
+  const handleClick = (event) => {
+    setAnchorEl(event.currentTarget);
+  };
+
   const handleClose = () => {
-    setOpen(false);
+    setAnchorEl(null);
   };
 
   return (
     <>
       <AppBar position="static">
-        <Toolbar style={{ backgroundColor: "#f25c54" }}>
+        <Toolbar sx={{ bgcolor: "#f25c54" }}>
```

Comment:
Consider repackaging this as an element at the App.js level so it renders on all pages instead or moving it into a component file that you can lodge into each page.

My suggestion is to implement it at the global level so you can learn how to pass props from each component properly